### PR TITLE
Issue 16758 create a dot avatar component

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: src/app/api/services/dot-gravatar-service.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    urls: string[];
+}
+@Injectable()
+export class DotGravatarService {
+    constructor(private jsonp: Jsonp) {}
+
+    /**
+     * Load the avatar url from a hash
+     *
+     * @param {string} hash
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-gravatar-service.ts]: remove `{}`

File: src/app/api/services/dot-gravatar-service.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+     * @returns {Observable<any>}
+     * @memberof GravatarService
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-gravatar-service.ts]: `DotGravatarService`

File: src/app/view/components/_common/dot-avatar/dot-avatar.component.scss [hunk 3]
```diff
@@ -4,16 +4,27 @@
...
+
+.avatar__dot {
+    height: 13px;
+    width: 13px;
+    border-radius: 50%;
+    display: inline-block;
+    position: absolute;
+    bottom: 2px;
+    right: 0px;
+    background-color: blueviolet;
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-avatar/dot-avatar.component.scss]: var?

File: src/app/view/components/_common/dot-avatar/dot-avatar.component.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+})
+export class DotAvatarComponent implements OnInit {
+    @Input()
+    url: string;
+
+    @Input()
+    showDot = false;
+
+    avatarPlaceholder: string;
+    avatarStyles;
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-avatar/dot-avatar.component.ts]: type?

File: src/app/view/components/_common/dot-avatar/dot-avatar.component.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            'line-height': value + 'px',
+            width: value + 'px'
+        };
+    }
+
+    get size(): number {
+        return this._size;
+    }
+
+    public getDotAvatar(label: string): any {
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-avatar/dot-avatar.component.ts]: doc

Comment (alfredo-dotcms) [reply src/app/view/components/_common/dot-avatar/dot-avatar.component.ts]: return `any`?

File: src/app/api/services/dot-gravatar-service.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    loadGravatarPhoto(hash: string): Observable<string> {
```
Comment (fmontes) [inline src/app/api/services/dot-gravatar-service.ts]:
```suggestion
    getGravatarPhoto(hash: string): Observable<string> {
```

File: src/app/shared/shared.module.ts [hunk 7]
```diff
@@ -31,12 +31,12 @@ const dotEventSocketURLFactory = () => {
...
 @NgModule({
     declarations: [],
-    imports: [CommonModule, DotDropdownModule, GravatarModule, MainNavigationModule],
+    imports: [CommonModule, DotDropdownModule, DotGravatarModule, MainNavigationModule],
     exports: [
         CommonModule,
         // Common Modules
         DotDropdownModule,
-        GravatarModule,
+        DotGravatarModule,
```
Comment (fmontes) [inline src/app/shared/shared.module.ts]: If we turn this into a module, we can just call this module inside the toolbar and we don't have to put it here in the share module.

File: src/app/view/components/_common/dot-avatar/dot-avatar.component.scss (hunk 3 above)
Comment (fmontes) [reply src/app/view/components/_common/dot-avatar/dot-avatar.component.scss]: primary.

File: src/app/view/components/_common/dot-avatar/dot-avatar.component.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    url: string;
+
+    @Input()
+    showDot = false;
+
+    avatarPlaceholder: string;
+    avatarStyles;
+
+    private _label: string;
+    private _size: number;
```
Comment (fmontes) [inline src/app/view/components/_common/dot-avatar/dot-avatar.component.ts]:
```suggestion
    private _size = 32;
```

File: src/app/view/components/_common/dot-avatar/dot-avatar.component.ts (hunk 5 above)
Comment (fmontes) [inline src/app/view/components/_common/dot-avatar/dot-avatar.component.ts]:
```suggestion
    getDotAvatar(label: string): any {
```

File: src/app/api/services/dot-gravatar-service.ts (hunk 1 above)
Comment (fmontes) [reply src/app/api/services/dot-gravatar-service.ts]: This is added by the doc extension.

**Comment by fmontes** [comment]
## Reorganization and rename
1. Break down the `dot-toolbar` elements into modules and add the dependencies to them
2. Rename the whole thing adding: `dot` prefix and `.component` sufix
3. Rename the files and folders accordingly
4. Rename the class names

**Pull request merged**